workflows/cloudify_system_workflows/snapshots/credentials.py

```diff
@@ -69,121 +70,136 @@ def _get_hosts():
                 for node in wctx.nodes
                 if is_compute(node)]
 
-    def _dump_agent_key(self, node_id, agent_key_path, archive_cred_path):
+    def _dump_agent_key(self, node_id, agent_key, archive_cred_path):
         """Copy an agent key from its location on the manager to the snapshot
         dump
         """
         os.makedirs(os.path.join(archive_cred_path, node_id))
-        source = os.path.expanduser(agent_key_path)
+        source = os.path.expanduser(agent_key)
         destination = os.path.join(archive_cred_path, node_id,
                                    self._CRED_KEY_NAME)
         ctx.logger.debug('Dumping credentials data, '
                          'copy from: {0} to {1}'
                          .format(source, destination))
-        shutil.copy(source, destination)
-
-    def _restore_agent_credentials(self,
-                                   dep_node_id,
-                                   dump_cred_dir,
-                                   restored_cred_dir,
-                                   agent_key_path_dict):
-        restored_agent_key_path = \
-            self._restore_agent_key_from_dump(dump_cred_dir,
-                                              restored_cred_dir,
-                                              dep_node_id)
-        db_agent_key_path = agent_key_path_dict[dep_node_id]
-
-        if os.path.isfile(db_agent_key_path):
-            self._handle_existing_agent_key_path(
-                restored_agent_key_path,
-                db_agent_key_path
-            )
-        else:
-            copy_file(restored_agent_key_path, db_agent_key_path)
-
-    @staticmethod
-    def _handle_existing_agent_key_path(restored_key_path,
-                                        db_key_path):
-        """Handle the case where the agent key already exists; either ignore
-        it if the contents are identical, or raise an error if they aren't
-
-        :param restored_key_path: The key path as restored from the dump
-        :param db_key_path: The key path as retrieved from the DB
-        """
-        with open(db_key_path) as key_file:
-                content_1 = key_file.read()
-        with open(restored_key_path) as key_file:
-            content_2 = key_file.read()
-        if content_1 != content_2:
-            raise NonRecoverableError(
-                'Agent key path already taken: {0}'.format(db_key_path)
-            )
-        ctx.logger.debug('Agent key path already exist: '
-                         '{0}'.format(db_key_path))
+        try:
+            shutil.copy(source, destination)
```

Comment:
It's against my religion (but seriously BAFP is generally considered more pythonic... and I know at the moment it's not really racey because nothing else *should* be happening at the same time on the manager, but I'd still rather try to set an example, and in any case this isn't *worse*.
This also means there's no need to write an "is it a file path?" function, with potential bugs in it. Just let the OS tell us if it's a file path.